subsys/usb/device/class/dfu/Kconfig

```diff
@@ -18,8 +18,18 @@ config USB_DEVICE_DFU_PID
 	help
 	  USB device product ID in DFU mode. MUST be configured by vendor.
 
+config USB_DFU_WILL_DETACH
+	bool "Generate detach-attach sequence on DFU detach"
+	default y
+	help
+	  Enabling this option makes the device responsible for detaching
+	  itself from the bus after the DFU_DETACH request. Select this
+	  for compatibility with host drivers that cannot issue USB reset.
+	  DFU fails on Windows with WinUSB driver if this is not enabled.
+
 config USB_DFU_DETACH_TIMEOUT
 	int
+	default 100 if USB_DFU_WILL_DETACH
```

Comment:
would be good to add `help` to this property. STH like: This determine time duration for which the device will stay detached during detach-attach sequence.

Reply:
This is hidden symbol, i.e. user cannot provide value, and I don't think it is necessary to describe it, especially that no only the value but also the meaning depends on `USB_DFU_WILL_DETACH`.

What you wrote is not what this timeout does. There is no artificial delay between the detach and attach. When `USB_DFU_WILL_DETACH` is enabled, the `USB_DFU_DETACH_TIMEOUT` determines the time after receiving `DFU_DETACH` request and when the device actually does detach. If `USB_DFU_WILL_DETACH` is disabled, it is the (defined by USB Device
Firmware Upgrade Specification Revision 1.1) timeout during which the host must initiate USB reset.